src/NServiceBus.AcceptanceTests/Recoverability/Retries/When_deferring_message_during_slr.cs

```diff
@@ -0,0 +1,112 @@
+﻿namespace NServiceBus.AcceptanceTests.Recoverability.Retries
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using System.Transactions;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using Features;
+    using NServiceBus.Config;
+    using NUnit.Framework;
+    using ScenarioDescriptors;
+
+    public class When_deferring_message_during_slr : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_not_commit_transaction()
+        {
+            await Scenario.Define<Context>(c => c.Id = Guid.NewGuid())
+                .WithEndpoint<Endpoint>(b => b.DoNotFailOnErrorMessages()
+                    .When((session, context) => session.SendLocal(new MessageToFail
+                    {
+                        Id = context.Id
+                    }))
+                 )
+                .WithEndpoint<ErrorSpy>()
+                .Done(c => c.MessageMovedToErrorQueue)
+                .Repeat(r => r.For<AllDtcTransports>())
+                .Should(c => Assert.That(c.TransactionStatuses, Is.All.Not.EqualTo(TransactionStatus.Committed)))
+                .Should(c => Assert.AreEqual(4, c.NumberOfProcessingAttempts))
+                .Run();
+        }
+
+        const string ErrorQueueName = "error_spy_queue";
+
+        class Context : ScenarioContext
+        {
+            public Guid Id { get; set; }
+            public bool MessageMovedToErrorQueue { get; set; }
+            public List<TransactionStatus> TransactionStatuses { get; } = new List<TransactionStatus>();
+            public int NumberOfProcessingAttempts { get; set; }
+        }
+
+        class Endpoint : EndpointConfigurationBuilder
+        {
+            public Endpoint()
+            {
+                EndpointSetup<DefaultServer>(config =>
+                {
+                    config.DisableFeature<FirstLevelRetries>();
+                    config.EnableFeature<SecondLevelRetries>();
+                    config.EnableFeature<TimeoutManager>();
+                    config.SendFailedMessagesTo(ErrorQueueName);
+                })
+                .WithConfig<SecondLevelRetriesConfig>(slrConfig =>
+                {
+                    slrConfig.NumberOfRetries = 3;
+                    slrConfig.TimeIncrease = TimeSpan.FromSeconds(1);
+                });
+            }
+
+            class FailingHandler : IHandleMessages<MessageToFail>
+            {
+                public Context TestContext { get; set; }
+
+                public Task Handle(MessageToFail message, IMessageHandlerContext context)
+                {
+                    if (message.Id == TestContext.Id)
+                    {
+                        TestContext.NumberOfProcessingAttempts++;
+
+                        Transaction.Current.TransactionCompleted += CaptureTransactionStatus;
+                    }
+
+                    throw new SimulatedException();
```

Comment:
I think you can move that inside the if statement. Then you can remove `DoNotFailOnErrorMessages()` and just assert that the message was moved to the error queue instead of having an error spy endpoint?